libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/api/TimeBasedNextVoiceInstructionsProvider.kt

```diff
@@ -0,0 +1,67 @@
+package com.mapbox.navigation.ui.voice.api
+
+import com.mapbox.api.directions.v5.models.VoiceInstructions
+import com.mapbox.navigation.utils.internal.logW
+
+internal class TimeBasedNextVoiceInstructionsProvider(
+    private val observableTimeSeconds: Int,
+) : NextVoiceInstructionsProvider {
+
+    override fun getNextVoiceInstructions(progress: RouteProgressData): List<VoiceInstructions> {
+        val legs = progress.route.legs() ?: return emptyList<VoiceInstructions>().also {
+            logW(
+                "Route does not contain legs, progress=$progress",
+                LOG_CATEGORY
+            )
+        }
+        var legSteps = legs.getOrNull(progress.legIndex)?.steps()
+        var currentStep = legSteps
+            ?.getOrNull(progress.stepIndex)
+            ?: return emptyList<VoiceInstructions>().also {
+                logW(
+                    "Route does not contain valid step, progress=$progress",
+                    LOG_CATEGORY
+                )
+            }
+
+        val voiceInstructions = mutableListOf<VoiceInstructions>()
+        var cumulatedTime = progress.stepDurationRemaining
```

Comment:
Maybe?
```suggestion
        var timeAheadOfProcessedVoiceInstructions = progress.stepDurationRemaining
```

Reply:
But it's exactly the time that has already been cumulated. It increases when I process new steps.

Reply:
I didn't understood what it does at the beginning so when I read and understood the code I decided to suggest a name which would help me understood the algorithm easier at the beginning. This is subjective topic, all I'm saying is that this would be more understandable for me. I'm okay if you don't agree and keep current name